clients/src/test/java/org/apache/kafka/clients/MetadataTest.java

```diff
@@ -199,19 +200,21 @@ public void testIgnoreLeaderEpochInOlderMetadataResponse() {
             MetadataResponse response = new MetadataResponse(struct, version);
             assertFalse(response.hasReliableLeaderEpochs());
             metadata.updateWithCurrentRequestVersion(response, false, 100);
+            assertFalse(metadata.hasReliableLeaderEpochs());
```

Comment:
Added test coverage L203 and L214, other changes are just side cleanups and signature refactoring in this file.